frontend/templates/product/sections/ReviewsSection/index.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+   Box,
+   Button,
+   Flex,
+   Grid,
+   GridItem,
+   Heading,
+   HStack,
+   Icon,
+   Link,
+   Progress,
+   Stack,
+   Tag,
+   Text,
+} from '@chakra-ui/react';
+import { Rating, RatingStar, RatingStarAppearance } from '@components/ui';
+import { useAppContext } from '@ifixit/app';
+import { PageContentWrapper } from '@ifixit/ui';
+import type { Product } from '@models/product';
+import { ProductVariant } from '@models/product';
+import React from 'react';
+import { FaShieldAlt } from 'react-icons/fa';
+import { useProductReviews } from '../../hooks/useProductReviews';
+
+const INITIAL_VISIBILE_REVIEWS = 3;
+
+export type ReviewsSectionProps = {
+   product: Product;
+   selectedVariant: ProductVariant;
+};
+
+export function ReviewsSection({
+   product,
+   selectedVariant,
+}: ReviewsSectionProps) {
+   const appContext = useAppContext();
+   const reviewsQuery = useProductReviews(product);
+   const [visibleReviewsCount, setVisibleReviewsCount] = React.useState(
+      INITIAL_VISIBILE_REVIEWS
+   );
+
+   const reviewsData = reviewsQuery.data;
+
+   const totalReviewsCount = reviewsData?.count ?? 0;
+
+   const reviewCountsByRating = React.useMemo(() => {
+      if (reviewsData?.groupedReviews == null) {
+         return [];
+      }
+      const reviewsCount = Object.keys(reviewsData.groupedReviews).reduce(
```

Comment:
Would a `.map()` have the same effect here? Might simplify things a little

Reply:
Yes, it can be simplified a bit with a map